### ISSUE TITLE
Testing plugin metadata

### ISSUE THREAD

**Issue by ling1726** [babel-utils/babel-plugin-tester#186]
It's possible to return metadata from the babel file transform result with a plugin. Is there a feature that can be used in the plugin tester to extract the metadata?

```ts
const babelFileResult = transform();
console.log('code', babelFileResult.code);
console.log('metadata, babelFileResult.metadata);
```

If not, would it be possible to have a feature that matches the returned metadata somehow?

This metadata can be set with a plugin and can return some useful information such as when the plugin is trying to extract some infomation from the source code while transforming it

**Comment by Xunnamius** [comment]
Interesting, and thank you for the example.

Is there any current documentation around the `metadata` property? The babel-core docs are seemingly incomplete in this regard and only describe `{ code, map, ast }` as the result of `transform(...)`, and the only potential docs reference to `metadata` I can find is [from the Babel 7 migration guide](https://babeljs.io/docs/v7-migration-api#babelcore).

~~I only ask because if it's an undocumented but stable (legacy?) property, I think a matcher specifically for `metadata` would be appropriate. But if it's an undocumented _new_ property, or there are a bunch of other undocumented stable properties, I'll opt for a more generic matcher instead.~~

I took a look at the types packages for [Babel 6](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/ad1ce9e540bcecba4dabe243c58467c49c562a47/types/babel-core/index.d.ts#L190-L196) versus [Babel 7](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/ad1ce9e540bcecba4dabe243c58467c49c562a47/types/babel__core/index.d.ts#L532-L549), and it seems like there are several undocumented stable properties. Perhaps a generic matcher property like `rawOutput: (output: BabelFileResult) => void` where you can bail out of the normal testing logic and make your own `expect` calls, e.g. `expect(output.metadata).toStrictEqual({ ... })`.

What do you think?

**Comment by ling1726** [comment]
Yeah returning the entire `transform` result would be a good idea regardless since it will let anyone test the entire stable return.

**Comment by Xunnamius** [comment]
🚀 This issue's resolution was included in babel-plugin-tester@12.0.0 ✨

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/babel-plugin-tester/v/12.0.0)
- [GitHub release](https://github.com/babel-utils/babel-plugin-tester/releases/tag/babel-plugin-tester%4012.0.0)
